### ISSUE TITLE
Spam Orders

### ISSUE THREAD

**Issue by thijskaspers** [enupal/stripe#363]
**Feature request**

It would be nice if there are some payment form events, like afterSubmission or beforeValidate, etc.

We are getting a lot of spam submissions and would like to implement ReCaptcha v3 to the payment form, but in order to do the server side verification to check if the captcha is correct, we need to check that after submission of the form. Enupal has a lot of events we can tap into, but most of them are for when Stripe has already been contacted? We want to edit the form after it's been validated but right before it does anything with Stripe.

### Additional info

- Craft version: 4+
- PHP version: 8

**Comment by siebird** [comment]
I have a few clients on Craft 3.8.x w/Stripe 4.1.2. 

Been battling spam with the same attack on multiple sites. 12k+ `test mode` orders overnight from the same email `testing@example.com`

We actually had to write a queue job to delete these orders. 

Why is the plugin processing orders and marking them as `test mode` when there is no test keys `env` variables on the live server?

**Comment by thijskaspers** [comment]
@siebird Seems we have the exact same spam, from testing@example.com and we also see "ghovjnjv" in the name field. Someone / a bot is targeting either Craft + Enupal Stripe, or Stripe forms in general.. Another website of ours that also uses Enupal Stripe has the spam issue too.

Would you mind sharing the queue job code?

And yes, we see payments with and without 'test mode', on production, where test mode is disabled.. and we also see the spam somehow changing the amount they have to pay, which is worrisome. Lastly, the payments are marked as "successful" in Enupal Stripe, even though we have the validate signature option enabled.

**Comment by andrelopez** [comment]
@siebird @thijskaspers could you please send back to us a zip file of the logs located at `storage/logs` to support@enupal.com. Also, if you can quickly describe how are you displaying the form, I will try to recreate this issue.

**Comment by siebird** [comment]
Emailed my logs w/an example page

**Comment by mufi-siebird** [comment]
@thijskaspers Here is the [zip file](https://github.com/enupal/stripe/files/12453416/customhelper.zip) of MODULE. You will need to:

1. add this code in `composer.json`
```
"autoload": {
    "psr-4": {
        "modules\\customHelper\\": "modules/customhelper/src/"
    }
},
```

2. add this in config/app.php`
```
return [
    'modules' => [
        'customhelper' => \modules\customHelper\CustomHelperModule::class,
    ],
    'bootstrap' => ['customhelper'],
];
```

3. run this command:
```
composer dump-autoload
```

4. Run queue from this command:
```
./craft customhelper/default/index
```

**Comment by siebird** [comment]
@andrelopez Any updates or hot fixes on this? 

Getting Stripe failed webhook notifications b/c of this too.

**Comment by siebird** [comment]
Another site got hit overnight.

**Comment by andrelopez** [comment]
Currently working on it! Will keep this thread updated

**Comment by andrelopez** [comment]
@thijskaspers could you please share your logs to continue debugging on this?

**Comment by andrelopez** [comment]
I was able to recreate this issue, we'll add a fix in place soon!

**Comment by andrelopez** [comment]
We just released Stripe Payments v5.3.0 for Craft 4 and V4.3.0 for Craft 3 with a fix to this issue. Thank you all for reporting and your help in debugging this issue.
